### PR TITLE
Explore: Removes Promise.All from runQueries thunk

### LINKED ISSUE #16956: Explore: Remove Promise.All from runQueries thunk

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Closes #16956

**Special notes for your reviewer**:

### CODE REVIEW

**Review by davkal — COMMENTED** [review]
Rx, yeah! Definitely takes some getting used to, but seems to make sense here.

I did some profiling testing in an incognito window (no extensions) with the 5s setting and thought I found a leak, but turns out the sample period was too short for the GC to run. So all good.

**Review by torkelo — APPROVED** [review]
nice!

File: public/app/features/explore/state/actions.ts [hunk 1]
```diff
@@ -638,7 +633,7 @@ function runQueriesForType(
     const { datasourceInstance, eventBridge, queries, queryIntervals, range, scanning } = getState().explore[exploreId];
     const datasourceId = datasourceInstance.meta.id;
     // Run all queries concurrently
-    const queryPromises = queries.map(async (query, rowIndex) => {
+    queries.map(async (query, rowIndex) => {
```
Comment (davkal) [inline public/app/features/explore/state/actions.ts]: Could be done as for-of since result is not used?

Comment (hugohaggmark) [reply public/app/features/explore/state/actions.ts]: Good catch, changed to for loop because we need the rowIndex.

**Comment by hugohaggmark** [comment]
> Rx, yeah! Definitely takes some getting used to, but seems to make sense here.
> 
> I did some profiling testing in an incognito window (no extensions) with the 5s setting and thought I found a leak, but turns out the sample period was too short for the GC to run. So all good.

Good to know you got my back @davkal and good to know there were no leaks.

**Pull request merged**